force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,48 @@
+trigger OpportunityTrigger on Opportunity (before delete, before update){
+    
+    if (Trigger.isBefore && Trigger.isUpdate) {
+        //Get Account Ids from Opportunities.
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity oppAccountIds : Trigger.new) {
+            accountIds.add(oppAccountIds.AccountId);
+        }
+        //Get CEO Contacts from Account Ids
+        List<Contact> ceoContacts = [
+            SELECT Id, Title, AccountId
+            FROM Contact
+            WHERE AccountId IN :accountIds AND Title = 'CEO'
+        ];
+        //Map Account Id with Contact.
+        Map<Id,Contact> primaryContacts = new Map<Id,Contact>();
+        for (Contact ceoContact : ceoContacts) {
+            primaryContacts.put(ceoContact.AccountId, ceoContact);
```

Comment:
a convention I really like when it comes to naming maps is  <value> by <key> so this would be contactsByAccountId. Makes it very readable.